src/XRP/reliable-submission-xrp-client.ts

```diff
@@ -114,20 +125,31 @@ export default class ReliableSubmissionXrpClient implements XrpClientDecorator {
     )
   }
 
+  private isMalformedTransactionStatus(transactionStatusCode: string): boolean {
+    return transactionStatusCode.startsWith('tem')
+  }
+
   private determineFinalResult(
     rawTransactionStatus: RawTransactionStatus,
   ): TransactionStatus {
+    const transactionStatusCode = rawTransactionStatus.transactionStatusCode
+
+    if (!transactionStatusCode) {
+      // Is this legit? What does this even mean? Is this just for testing purposes?
+      throw new Error('no')
+    }
+
     // Return pending if the transaction is not validated.
     if (!rawTransactionStatus.isValidated) {
-      return TransactionStatus.Pending
-    } else {
-      const transactionStatus = rawTransactionStatus.transactionStatusCode?.startsWith(
-        'tes',
-      )
-        ? TransactionStatus.Succeeded
-        : TransactionStatus.Failed
-      return transactionStatus
+      return this.isMalformedTransactionStatus(transactionStatusCode)
+        ? TransactionStatus.MalformedTransaction
+        : TransactionStatus.Pending
```

Comment:
there seem to be other cases here too, but i only care about MalformedTranscation